File: 프로그래머스/2/42586. 기능개발/기능개발.js
function solution(progresses, speeds) {
    var answer = [];
    for (let i = 0; i < progresses.length; i++) {
        speeds[i] = Math.ceil((100-progresses[i])/speeds[i])
    };
    let cnt = 1;
    for (let i = 0; i < progresses.length; i++) {
        if (i+1<progresses.length) {
            if (speeds[i] >= speeds[i+1]) {
                cnt++;
                speeds[i+1] = speeds[i]
            } else {
                answer.push(cnt)
                cnt = 1
            }
        } else {
            answer.push(cnt)
        }
    }
    return answer;
}

File: 프로그래머스/2/42586. 기능개발/기능개발.py
import math
def solution(progresses, speeds):
    len_p = len(progresses)
    answer = []
    for i in range(len_p):
        speeds[i] = math.ceil((100-progresses[i])/speeds[i])
        # [7, 3, 9]
    cnt = 1
    for i in range(len_p):
        if i + 1 < len_p:
            if speeds[i] >= speeds[i+1]:
                cnt += 1
                speeds[i+1] = speeds[i]
            else:
                answer.append(cnt)
                cnt = 1
        else:
            answer.append(cnt)
            
    return answer